syserol/regression.py

```diff
@@ -28,3 +30,35 @@ def patientComponents(nComp=1):
     print(np.sqrt(r2_score(Y, Y_pred)))
 
     return Y, Y_pred
+
+def function_elastic_net(function='ADCC'):
+    #Import Luminex, Luminex-IGG, Function, and Glycan into DF
+    df = importLuminex()
+    lum = df.pivot(index='subject', columns = 'variable', values = 'value')
+    glycan, df2 = importGlycan()
+    glyc = df2.pivot(index='subject', columns = 'variable', values = 'value')
+    func = importFunction()
+    igg = importIGG()
+    igg = igg.pivot(index='subject', columns = 'variable', values = 'value')
+    data_frames = [lum, glyc, func, igg]
+    df_merged = reduce(lambda  left,right: pd.merge(left,right,on=['subject'],
+                                                how='inner'), data_frames)
+    df_merged = df_merged.dropna()
+
+    #separate dataframes
+    df_func = df_merged[["ADCD", "ADCC", "ADNP", "CD107a", "IFNy", "MIP1b"]]
+    df_variables = df_merged.drop(['subject','ADCD', 'ADCC', 'ADNP', 'CD107a', 'IFNy', 'MIP1b'], axis = 1)
+
+    #perform regression
+    Y = df_func[function]
+    X = df_variables
+    Y_pred = np.empty(Y.shape)
+
+    Y_pred = cross_val_predict(ElasticNetCV(normalize=True, max_iter=10000), X, Y, cv=len(Y))
```

Comment:
@murphymadeleine21 so the issue is we're doing cross-val inside of cross-val, so with ~180 subjects this results in fitting ~32,000 models. Instead of using `cross_val_predict`, `ElasticNetCV` has a member variable `alpha_` which gives the alpha it picked as best. You can then run `cross_val_predict` with `sklearn.linear_model.ElasticNet` and this best picked alpha to get the cross validation results. This should reduce the number of models fit to ~360. Finally, use `cv=10` instead of `cv=len(Y)`. This will then require 10 models fit instead of ~180 during the second half.